aten/src/ATen/core/TensorImpl.h

```diff
@@ -847,6 +847,8 @@ struct CAFFE2_API TensorImpl : public c10::intrusive_ptr_target {
    * preserves the DeviceType of the source tensor (so, e.g., if you allocate
    * a tensor on CPU and then CopyFrom a CUDA tensor, that will to a
    * CUDA-to-CPU transfer).
+   *
+   * If the function is invoked without `context` the copy would be synchronous
```

Comment:
I'm actually not sure how this follows from the definition of this method below. Don't you need `FinishDeviceComputation()` to synchronize the copy? I don't see this method call in the body below.

Reply:
Ah! The destructor of Context object calls FinishDeviceComputation. Context is basically RAII if it's created on the fly :)

Reply:
Great, thank you :)